dsc/src/main.rs

```diff
@@ -341,56 +351,38 @@ impl DscInfo {
             }
         }
 
-        let ds_state = vec![
-            DownstairsState::Stopped,
-            DownstairsState::Stopped,
-            DownstairsState::Stopped,
-        ];
-        let ds_pid = vec![None, None, None];
+        let mut ds_state = Vec::new();
+        let mut ds_pid = Vec::new();
+
+        for _ in 0..region_count {
+            ds_state.push(DownstairsState::Stopped);
+            ds_pid.push(None);
+        }
 
         // If we only have one region directory, then use that for all
         // three downstairs.  If we have three, then each downstairs can
```

Comment:
`If we have N` here?

Reply:
Ah yeah, I'll update the comment before merging.